components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -440,7 +443,11 @@ private OAuth2AccessTokenRespDTO generateNewAccessToken(OAuthTokenReqMessageCont
 
         Timestamp timestamp = new Timestamp(new Date().getTime());
         updateMessageContextToCreateNewToken(tokReqMsgCtx, consumerKey, existingTokenBean, timestamp);
-        executePreIssueAccessTokenActions(tokReqMsgCtx);
+        ActionExecutionStatus<?> executionStatus = executePreIssueAccessTokenActions(tokReqMsgCtx);
+        if (executionStatus != null && (executionStatus.getStatus() == ActionExecutionStatus.Status.FAILED ||
```

Comment:
IMO, execution status should not be null
May be we can add a unit test to verify this contract is hold always
And if course if unit test is not present better to have this check than not have it

Reply:
Will be removing this check once the unit tests are added. Added a TODO.

Reply:
Based on the method contract of "executePreIssueAccessTokenActions" null value is a possible value that can be returned as it has a check on whether to invoke the extensions or not. In that case null check is needed.

Do the needful accordingly

Reply:
Updated with c9f7af2d45225346605e5e5b9a0d7953bc642ff2